persistence/src/main/java/com/epam/esm/persistence/dao/impl/CertificateDAOImpl.java

```diff
@@ -58,30 +60,34 @@ public Certificate create(Certificate certificate) throws DAOSQLException {
     }
 
     @Override
-    public Certificate read(int id) throws DAOSQLException {
-        return template.query(CertificateQueries.SELECT_FROM_CERTIFICATE.getValue()
-                        .concat(CertificateQueries.WHERE_ID.getValue()),
-                certificateMapper, id).stream().findFirst().get();
+    public Certificate read(int id) {
+        try {
+            return template.queryForObject(CertificateQueries.SELECT_FROM_CERTIFICATE.getValue()
+                            .concat(CertificateQueries.WHERE_ID.getValue()),
+                    certificateMapper, id);
+        } catch (EmptyResultDataAccessException e) {
+            return null;
```

Comment:
could you please clarify why we need to return null in catch block?

Reply:
Discussed,